include/seqan3/range/container/concept.hpp

```diff
@@ -127,15 +171,21 @@ concept bool sequence_concept = requires (type val, type val2)
 
     // modify container
 //TODO: how do you model this?
-//     { val.emplace(typename type::const_iterator{}, ?                                   } -> typename type::iterator;
-    { val.insert(val.cbegin(), val2.front())                                           } -> typename type::iterator;
-    { val.insert(val.cbegin(), typename type::value_type{})                            } -> typename type::iterator;
-    { val.insert(val.cbegin(), typename type::size_type{}, typename type::value_type{})} -> typename type::iterator;
-    { val.insert(val.cbegin(), val2.begin(), val2.end())                               } -> typename type::iterator;
-//TODO this fails on std::string, although it should work
-//     { val.insert(val.cbegin(), std::initializer_list<typename type::value_type>{})    } -> typename type::iterator;
-    { val.erase(val.cbegin())                                                          } -> typename type::iterator;
-    { val.erase(val.cbegin(), val.cend())                                              } -> typename type::iterator;
+    // { val.emplace(typename type::const_iterator{}, ?                                   } -> typename type::iterator;
+
+    { val.insert(val.begin(), val2.front())                                            } -> typename type::iterator;
+    { val.insert(val.begin(), typename type::value_type{})                             } -> typename type::iterator;
+    // because of a travis bug we can't assume typename type::iterator as return type
+    { val.insert(val.begin(), typename type::size_type{}, typename type::value_type{}) };
+    { val.insert(val.begin(), val2.begin(), val2.end())                                };
+    //TODO should return type::iterator on strings (https://gcc.gnu.org/bugzilla/show_bug.cgi?id=83328)
+    { val.insert(val.begin(), std::initializer_list<typename type::value_type>{})      };
+    { val.erase(val.begin())                                                           } -> typename type::iterator;
+    { val.erase(val.begin(), val.end())                                                } -> typename type::iterator;
+
+    // workaround a travis bug where insert/erase can't take a const iterator, e.g. cbegin()
+    requires detail::sequence_concept_modified_by_const_iterator_bug || detail::sequence_concept_modified_by_const_iterator<type>;
```

Comment:
Please add a check, that this is only tested if the queried type is in fact a std::basic_string.